presto-spi/src/main/java/io/prestosql/spi/type/TDigestType.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.airlift.slice.Slice;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.connector.ConnectorSession;
+
+public class TDigestType
+        extends AbstractVariableWidthType
+{
+    public static final TDigestType TDIGEST = new TDigestType();
+
+    @JsonCreator
+    public TDigestType()
+    {
+        super(new TypeSignature(StandardTypes.TDIGEST), Slice.class);
```

Comment:
We can use TDigest as the native type instead of serializing and deserializing. See, for instance, LongTimestampType